src/txkube/_authentication.py

```diff
@@ -202,7 +202,8 @@ def creatorForNetloc(self, hostname, port):
 
         :see: ``twisted.web.iweb.IPolicyForHTTPS``
         """
-        hostname = hostname.decode("ascii")
+        if isinstance(hostname, bytes):
+            hostname = hostname.decode("ascii")
```

Comment:
Twisted interface docs say `hostname` is `bytes`.  So either Twisted docs are broken or Twisted code is broken.